trunk/source/Library-TraceCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/tracecheckerutils/singletracecheck/AnnotateAndAsserter.java

```diff
@@ -51,6 +52,7 @@ public class AnnotateAndAsserter<L extends IAction> {
 
 	protected final ManagedScript mMgdScriptTc;
 	protected final NestedWord<L> mTrace;
+	protected final List<?> mControlConfigurationSequence;
```

Comment:
For concurrent programs we do not have sequences of locations but sequence of *SomethingElse*. We do not yet know what *SomethingElse* should be.
* One good choice for *SomethingElse* is a set of locations, one location per active thread.
* Another choice might be that we additionally states of a monitor automaton.

Reply:
`configuration` just seems awfully close to the mixing of concepts we want to avoid. 
For GemCutter, I would argue that the pair of sets of location and the set of letters is the location of the explicit product automaton, so here `location` would be ok. I think that `location` is in general closer to the "something" we want to describe than `configuration`. But right now I don't know a better name.